### ISSUE TITLE
issue: i am having an issue while  deploying the contract to local ganache

### ISSUE THREAD

**Issue by ritesh798** [PatrickAlphaC/ethers-simple-storage-fcc#18]
hi all, at 7:08 hrs when I am trying to deploy my contract to local ganache it gets stuck there as my promise is not getting resolved 
... solidity
const ethers = require("ethers");
const fs = require("fs-extra");

async function main() {
  //RPC SERVER
  //HTTP://127.0.0.1:8545

  const provider = new ethers.providers.JsonRpcProvider(
    "HTTP://172.22.192.1:7545"
  );
  const wallet = new ethers.Wallet(
    "76fe5a715ceefaaa33f4bc5732d89f0b6057a76605d478e199b865fb162c7729",
    provider
  );
  const abi = fs.readFileSync("./SimpleStorage_sol_SimpleStorage.abi", "utf8");
  const binary = fs.readFileSync(
    "./SimpleStorage_sol_SimpleStorage.bin",
    "utf8"
  );
  const contractFactory = new ethers.ContractFactory(abi, binary, wallet);
  console.log("deploying please wait");
  const contract = await contractFactory.deploy();
  console.log(contract);
}

main()
  .then(() => process.exit(0))
  .catch((error) => {
    console.error(error);
    process.exit(1);
  });

...
here I'm copying my bash terminal:
...
ritesh@LAPTOP-TI9KIBT3:~/hardhatfcc/ethers-simple-storage$ node deploy.js
deploying please wait
...

**Comment by yatinkhandelwal** [comment]
yeah I'm getting the same error. I also tried vEthernet wsl and all interfaces option as well

**Comment by gabucino** [comment]
Try the solution by Izmanovichin from the other topic:
https://github.com/PatrickAlphaC/ethers-simple-storage-fcc/issues/17

> yeah I'm getting the same error. I also tried vEthernet wsl and all interfaces option as well

Have you refreshed the RPC server and the private key after changing to vEthernet wsl? The address changes, when you modify it - at least the was the issue in my case.

**Comment by ritesh798** [comment]
> yeah I'm getting the same error. I also tried vEthernet wsl and all interfaces option as well

I was unable to resolve it . but instead I tried with hardhat which Patrick told in one of the discussions. go check out discussions and there you will find how to start rpc with hardhat. #90 discussion.

**Comment by yatinkhandelwal** [comment]
@ritesh798 what you're referring to is it present in the video itself?

What if I leave this current state of not being able to deploy using node deploy.js , can I still continue with the video or will I face some issues?

**Comment by yatinkhandelwal** [comment]
@gabucino yeah I did that and it Didn't work.
Can I move forward with that video or should I continue looking for a solution to this cause its been 2-3 days since I've been stuck on this..

**Comment by ritesh798** [comment]
> @gabucino yeah I did that and it Didn't work. Can I move forward with that video or should I continue looking for a solution to this cause its been 2-3 days since I've been stuck on t

its not resolved inn video go to github discussions section there in discussion #90 Patrick gave full guide to start hardhat rpc. you just have to  start hardhat rpc in terminal and keep that terminal alive . Add new terminal for deployment . you will get hardhar rpc endpoint and private key on terminal replace them with genache and it will run.

**Comment by yatinkhandelwal** [comment]
@ritesh798 I think this discussion does not exist cause I'm not able to find it.

**Comment by M-Marcel** [comment]
> @gabucino yeah I did that and it Didn't work. Can I move forward with that video or should I continue looking for a solution to this cause its been 2-3 days since I've been stuck on this..

@yatinkhandelwal try this
https://github.com/PatrickAlphaC/ethers-simple-storage-fcc/issues/18#issuecomment-1166598141

**Comment by deepakdubey8756** [comment]
Hey @ritesh798  why are you using `"HTTP://172.22.192.1:7545"` instead of `http://127.0.0.1:7545`.

**Comment by cjayjoo** [comment]
Once these things were updated, I was able to deploy .

https://github.com/smartcontractkit/full-blockchain-solidity-course-js/discussions/34#discussioncomment-2846436

```
yarn add ganache
```

**Comment by hessinebkl** [comment]
i found a solution for me i was using WIFI, so i just changed the ganache server settings to Wi-Fi and it worked fine, :)

**Comment by ritesh798** [comment]
> i found a solution for me i was using WIFI, so i just changed the ganache server settings to Wi-Fi and it worked fine, :)

this might work because I was using wifi too. thanks

**Comment by kevinabraham23** [comment]
if ur using wsl ubuntu on a windows system it wont work
https://github.com/smartcontractkit/full-blockchain-solidity-course-js/blob/main/chronological-updates.md#using-the-wsl-endpoint
just go to this link and try any one of these steps and u should be fine :)

**Comment by A5TA** [comment]
Hey I had a similar issue but all I needed to do was add parameters for the gas limit and price for it to run
```js 
const contract = await contractFactory.deploy({ gasPrice: 100000000000, gasLimit: 1000000 })
```

**Comment by avi94** [comment]
Solution -> https://github.com/PatrickAlphaC/ethers-simple-storage-fcc/issues/85#issuecomment-1712202485
